src/main/java/com/nvidia/spark/rapids/jni/DateTimeUtils.java

```diff
@@ -0,0 +1,99 @@
+   * the result as in Julian calendar, then compute the number of days or microseconds since the
+   * epoch from that Julian local date-time.
+   * <p>
+   * This is to match with Apache Spark's `localRebaseGregorianToJulianDays` and
+   * `rebaseGregorianToJulianMicros` functions with timezone fixed to UTC.
+   *
+   * @param input The input column
+   * @return A new column with the rebase applied
+   */
+  public static ColumnVector rebaseGregorianToJulian(ColumnView input) {
+    return new ColumnVector(rebaseGregorianToJulian(input.getNativeView()));
+  }
+
+  /**
+   * Convert the given timestamps as a number of days or microseconds since the epoch instant
+   * 1970-01-01T00:00:00Z to a local date-time in Julian calendar, reinterpreting the result
+   * as in Proleptic Gregorian calendar, then compute the number of days or microseconds since the
+   * epoch from that Gregorian local date-time.
+   * <p>
+   * This is to match with Apache Spark's `localRebaseJulianToGregorianDays` and
+   * `rebaseJulianToGregorianMicros` functions with timezone fixed to UTC.
+   *
+   * @param input The input column
+   * @return A new column with the rebase applied
+   */
+  public static ColumnVector rebaseJulianToGregorian(ColumnView input) {
+    return new ColumnVector(rebaseJulianToGregorian(input.getNativeView()));
+  }
+
+  /**
+   * Truncate the given date or timestamp to the unit specified by the format string.
+   * <p>
+   * The input date/time must be of type TIMESTAMP_DAYS or TIMESTAMP_MICROSECONDS, and the format
+   * be of type STRING. In addition, the format strings are case-insensitive.
+   * <p>
+   * For TIMESTAMP_DAYS, the valid format are:<br>
+   *  - {@code "YEAR", "YYYY", "YY"}: truncate to the first date of the year.<br>
+   *  - {@code "QUARTER"}: truncate to the first date of the quarter.<br>
+   *  - {@code "MONTH", "MM", "MON"}: truncate to the first date of the month.<br>
+   *  - {@code "WEEK"}: truncate to the Monday of the week.<br>
+   * <br/>
+   * For TIMESTAMP_MICROSECONDS, the valid format are:<br>
+   *  - {@code "YEAR", "YYYY", "YY"}: truncate to the first date of the year.<br>
+   *  - {@code "QUARTER"}: truncate to the first date of the quarter.<br>
+   *  - {@code "MONTH", "MM", "MON"}: truncate to the first date of the month.<br>
+   *  - {@code "WEEK"}: truncate to the Monday of the week.<br>
+   *  - {@code "DAY", "DD"}: zero out the time part.<br>
+   *  - {@code "HOUR"}: zero out the minute and second with fraction part.<br>
+   *  - {@code "MINUTE"}: zero out the second with fraction part.<br>
+   *  - {@code "SECOND"}: zero out the second fraction part.<br>
+   *  - {@code "MILLISECOND"}: zero out the microseconds.<br>
+   *  - {@code "MICROSECOND"}: keep everything.<br>
+   *
+   * @param datetime The input date/time
+   * @param format The time component to truncate to
+   * @return The truncated date/time
+   */
+  public static ColumnVector truncate(ColumnView datetime, ColumnView format) {
```

Comment:
Could we have a version that supports a scalar format? I just think that will be the most common use case and if we can save memory and time by not exploding the scalar into a column I think that would be great.

I think that can be done as a follow on issue if you want.

Reply:
I thought about that, however, either parameter can be Scalar thus if doing so we will have to implement 4 JNI overloads for this function. Instead, the code supports columns in which any of them can have one row and can achieve the same performance. 

In plugin code, I just "convert"  Scalar into a column of one row. Such conversion should not be very expensive. But I can add more overload to support Scalar if that can help anything better.

Reply:
Yes either can be a scalar, but fmt is the one that is really common to be a scalar. Most of the time you want to truncate to a very specific value so you know how to deal with the resulting timestamp/date.